discovery-provider/src/tasks/tracks.py

```diff
@@ -172,6 +173,97 @@ def update_remixes_table(session, track_record, track_metadata):
                     )
                     session.add(remix)
 
+
+def update_track_routes_table(session, track_record, track_metadata):   
+    # Get the title slug, and set the new slug to that
+    # (will check for conflicts later)
+    new_track_slug_title = (
+        helpers.create_track_slug(track_record.title)
+    )
+    new_track_slug = new_track_slug_title
+
+    # Find the current route for the track
+    prev_track_route_record = (
+        session
+        .query(TrackRoute)
+        .filter(TrackRoute.track_id == track_record.track_id,
+                TrackRoute.is_current == True)
+        .one_or_none()
+    )
+
+    if prev_track_route_record is not None:
+        if prev_track_route_record.title_slug == new_track_slug_title:
+            # If the title slug hasn't changed, we have no work to do
+            return
+        else:
+            # The new route will be current
+            prev_track_route_record.is_current = False
+
+    # Check for conflicts Find the max conflict ID for the track slug
+    max_collision_id = (
+        session
+        .query(functions.max(TrackRoute.collision_id))
+        .filter(TrackRoute.title_slug == new_track_slug_title,
+                TrackRoute.owner_id == track_record.owner_id)
+        .one_or_none()
+    )[0]
+
+    new_collision_id = 0
+    has_collisions = False
+    if max_collision_id is not None:
+        has_collisions = True
+        new_collision_id = max_collision_id
+    while has_collisions:
+        # If there is an existing track by the user with that slug,
+        # then we need to append the collision number to the slug
+        new_collision_id += 1
+        new_track_slug = helpers.create_track_slug(
+            track_metadata["title"],
+            new_collision_id)
+
+        # Check for new collisions after making the new slug
```

Comment:
@raymondjacobson @piazzatron I've been racking my brain trying to find a better method than this to deal with repeated conflicts. Let me know if this doesn't make sense or if you have ideas. We can hop on a call early next week?